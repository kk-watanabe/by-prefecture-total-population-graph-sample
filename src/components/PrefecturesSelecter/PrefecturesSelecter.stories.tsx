import { prefecturesMock } from "../__mock__";
import { PrefecturesSelecter } from "./PrefecturesSelecter";

import type { ComponentStory, ComponentMeta } from "@storybook/react";

export default {
  title: "components/PrefecturesSelecter",
  component: PrefecturesSelecter,
  argTypes: {
    onChange: { action: "changed" },
  },
} as ComponentMeta<typeof PrefecturesSelecter>;

const Template: ComponentStory<typeof PrefecturesSelecter> = (args) => <PrefecturesSelecter {...args} />;

export const Default = Template.bind({});
Default.args = {
  prefectures: prefecturesMock.result,
};
