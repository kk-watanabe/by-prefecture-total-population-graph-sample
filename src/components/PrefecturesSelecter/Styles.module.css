.prefectures-selecter {
  display: flex;
  flex-direction: column;

  gap: 20px;
}

.prefectures-selecter__list {
  display: flex;
  flex-wrap: wrap;

  gap: 12px;
}

.prefectures-selecter__button {
  padding: 10px 16px 8px;
  border: 1px solid var(--color-gray300);
  border-radius: 4px;
  background-color: var(--color-gray100);
  color: var(--color-gray700);
  font-size: var(--fs-xs);
  line-height: 1;
  cursor: pointer;
  transition: background-color 0.3s ease;

  appearance: none;
}

.prefectures-selecter__button:hover {
  background-color: var(--color-gray200);
}
