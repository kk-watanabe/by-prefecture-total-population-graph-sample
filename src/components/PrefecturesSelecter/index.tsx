/* istanbul ignore next */
export * from "./PrefecturesSelecter";
