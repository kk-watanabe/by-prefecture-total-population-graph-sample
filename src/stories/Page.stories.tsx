import React from "react";

import { Page } from "./Page";

import type { ComponentStory, ComponentMeta } from "@storybook/react";

export default {
  title: "Example/Page",
  component: Page,
  parameters: {
    // More on Story layout: https://storybook.js.org/docs/react/configure/story-layout
    layout: "fullscreen",
  },
} as ComponentMeta<typeof Page>;

const Template: ComponentStory<typeof Page> = (args) => <Page {...args} />;

export const LoggedOut = Template.bind({});

export const LoggedIn = Template.bind({});
