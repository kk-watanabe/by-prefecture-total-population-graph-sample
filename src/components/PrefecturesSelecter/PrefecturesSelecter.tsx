import { useState, useCallback } from "react";

import { Checkbox } from "@/components/Checkbox";

import { Pref } from "@/api/prefectures/type";
import {
  hokkaidoAndTouhokuCodes,
  kantoCodes,
  tyubuCodes,
  kinkiCodes,
  tyugokuAndshikokuCodes,
  kyusyuAndOkinawaCodes,
  getRangePrefectures,
  getPrefColor,
  getPrefecturesWithValueAddOrReduce,
} from "@/utils/prefectures-util";

import styles from "./Styles.module.css";

export interface PrefecturesSelecterProps {
  prefectures: Pref[];
  onChange: (selectedPrefectures: Pref[]) => void;
}

export const PrefecturesSelecter = (props: PrefecturesSelecterProps) => {
  const { prefectures, onChange } = props;
  const [selectedPrefectures, setSelectedPrefectures] = useState<Pref[]>(
    prefectures.length > 0 ? [prefectures[0]] : []
  );

  const handleChnage = useCallback(
    (pref: Pref) => {
      const result = getPrefecturesWithValueAddOrReduce(selectedPrefectures, pref);

      setSelectedPrefectures(result);
      onChange(result);
    },
    [onChange, selectedPrefectures]
  );

  const checked = useCallback(
    (prefCode: number) => selectedPrefectures.some((p) => p.prefCode === prefCode),
    [selectedPrefectures]
  );

  const handleClick = useCallback(
    (codes: number[]) => {
      const rangePrefectures = getRangePrefectures(prefectures, codes);
      const result = new Set([...selectedPrefectures, ...rangePrefectures]);

      setSelectedPrefectures([...result]);
      onChange([...result]);
    },
    [onChange, selectedPrefectures, prefectures]
  );

  const allSelectedClick = useCallback(() => {
    setSelectedPrefectures(prefectures);
    onChange(prefectures);
  }, [onChange, prefectures]);

  const resetSelectedClick = useCallback(() => {
    setSelectedPrefectures([]);
    onChange([]);
  }, [onChange]);

  return (
    <div className={styles["prefectures-selecter"]}>
      <ul className={styles["prefectures-selecter__list"]}>
        <li>
          <button type="button" className={styles["prefectures-selecter__button"]} onClick={allSelectedClick}>
            全選択
          </button>
        </li>
        <li>
          <button type="button" className={styles["prefectures-selecter__button"]} onClick={resetSelectedClick}>
            全解除
          </button>
        </li>
      </ul>

      <ul className={styles["prefectures-selecter__list"]}>
        <li>
          <button
            type="button"
            className={styles["prefectures-selecter__button"]}
            onClick={() => handleClick(hokkaidoAndTouhokuCodes)}
          >
            北海道・東北を選択
          </button>
        </li>
        <li>
          <button
            type="button"
            className={styles["prefectures-selecter__button"]}
            onClick={() => handleClick(kantoCodes)}
          >
            関東を選択
          </button>
        </li>
        <li>
          <button
            type="button"
            className={styles["prefectures-selecter__button"]}
            onClick={() => handleClick(tyubuCodes)}
          >
            中部を選択
          </button>
        </li>
        <li>
          <button
            type="button"
            className={styles["prefectures-selecter__button"]}
            onClick={() => handleClick(kinkiCodes)}
          >
            近畿を選択
          </button>
        </li>
        <li>
          <button
            type="button"
            className={styles["prefectures-selecter__button"]}
            onClick={() => handleClick(tyugokuAndshikokuCodes)}
          >
            中国・四国を選択
          </button>
        </li>
        <li>
          <button
            type="button"
            className={styles["prefectures-selecter__button"]}
            onClick={() => handleClick(kyusyuAndOkinawaCodes)}
          >
            九州・沖縄を選択
          </button>
        </li>
      </ul>

      <ul className={styles["prefectures-selecter__list"]}>
        {prefectures.map((p, i) => (
          <li key={p.prefCode}>
            <Checkbox
              id={`item${i}`}
              colorCode={getPrefColor(p.prefCode) ?? ""}
              value={p.prefCode}
              label={p.prefName}
              checked={checked(p.prefCode)}
              onChange={() => handleChnage(p)}
            />
          </li>
        ))}
      </ul>
    </div>
  );
};
