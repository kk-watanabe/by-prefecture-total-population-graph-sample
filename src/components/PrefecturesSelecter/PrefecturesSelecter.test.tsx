import { composeStories } from "@storybook/testing-react";
import { render, screen, waitFor, act } from "@testing-library/react";
import { vi } from "vitest";

import * as stories from "./PrefecturesSelecter.stories";

const { Default } = composeStories(stories);

describe("snapshots", () => {
  it("Readner Default components", () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("onChange", () => {
  it("return Pref[]", async () => {
    const onChangeSpy = vi.fn();
    render(<Default onChange={onChangeSpy} />);

    await act(async () => {
      const checkboxes = screen.getAllByRole("checkbox");
      checkboxes.at(3)?.click();

      await waitFor(() => {
        expect(onChangeSpy).toHaveBeenCalled();
        expect(onChangeSpy).toBeCalledWith([
          {
            prefCode: 1,
            prefName: "北海道",
          },
          {
            prefCode: 16,
            prefName: "富山県",
          },
        ]);
      });
    });
  });
});
