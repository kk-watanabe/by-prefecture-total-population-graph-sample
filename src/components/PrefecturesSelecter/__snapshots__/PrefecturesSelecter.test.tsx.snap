// Vitest Snapshot v1

exports[`snapshots > Readner Default components 1`] = `
<div
  class="_prefectures-selecter_ueqfg_1"
>
  <ul
    class="_prefectures-selecter__list_ueqfg_8"
  >
    <li>
      <button
        class="_prefectures-selecter__button_ueqfg_15"
        type="button"
      >
        全選択
      </button>
    </li>
    <li>
      <button
        class="_prefectures-selecter__button_ueqfg_15"
        type="button"
      >
        全解除
      </button>
    </li>
  </ul>
  <ul
    class="_prefectures-selecter__list_ueqfg_8"
  >
    <li>
      <button
        class="_prefectures-selecter__button_ueqfg_15"
        type="button"
      >
        北海道・東北を選択
      </button>
    </li>
    <li>
      <button
        class="_prefectures-selecter__button_ueqfg_15"
        type="button"
      >
        関東を選択
      </button>
    </li>
    <li>
      <button
        class="_prefectures-selecter__button_ueqfg_15"
        type="button"
      >
        中部を選択
      </button>
    </li>
    <li>
      <button
        class="_prefectures-selecter__button_ueqfg_15"
        type="button"
      >
        近畿を選択
      </button>
    </li>
    <li>
      <button
        class="_prefectures-selecter__button_ueqfg_15"
        type="button"
      >
        中国・四国を選択
      </button>
    </li>
    <li>
      <button
        class="_prefectures-selecter__button_ueqfg_15"
        type="button"
      >
        九州・沖縄を選択
      </button>
    </li>
  </ul>
  <ul
    class="_prefectures-selecter__list_ueqfg_8"
  >
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item0"
      >
        <input
          checked=""
          class="_checkbox__input_1py7j_10"
          id="item0"
          type="checkbox"
          value="1"
        />
        <span
          class="_checkbox__box_1py7j_19 "
          style="background-color: rgb(91, 33, 182); border-color: #5b21b6;"
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 _checkbox__label--checked_1py7j_48"
        >
          北海道
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item1"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item1"
          type="checkbox"
          value="2"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          青森県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item2"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item2"
          type="checkbox"
          value="3"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          岩手県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item3"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item3"
          type="checkbox"
          value="4"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          宮城県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item4"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item4"
          type="checkbox"
          value="5"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          秋田県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item5"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item5"
          type="checkbox"
          value="6"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          山形県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item6"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item6"
          type="checkbox"
          value="7"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          福島県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item7"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item7"
          type="checkbox"
          value="8"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          茨城県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item8"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item8"
          type="checkbox"
          value="9"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          栃木県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item9"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item9"
          type="checkbox"
          value="10"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          群馬県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item10"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item10"
          type="checkbox"
          value="11"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          埼玉県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item11"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item11"
          type="checkbox"
          value="12"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          千葉県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item12"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item12"
          type="checkbox"
          value="13"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          東京都
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item13"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item13"
          type="checkbox"
          value="14"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          神奈川県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item14"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item14"
          type="checkbox"
          value="15"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          新潟県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item15"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item15"
          type="checkbox"
          value="16"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          富山県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item16"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item16"
          type="checkbox"
          value="17"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          石川県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item17"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item17"
          type="checkbox"
          value="18"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          福井県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item18"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item18"
          type="checkbox"
          value="19"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          山梨県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item19"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item19"
          type="checkbox"
          value="20"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          長野県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item20"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item20"
          type="checkbox"
          value="21"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          岐阜県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item21"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item21"
          type="checkbox"
          value="22"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          静岡県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item22"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item22"
          type="checkbox"
          value="23"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          愛知県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item23"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item23"
          type="checkbox"
          value="24"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          三重県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item24"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item24"
          type="checkbox"
          value="25"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          滋賀県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item25"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item25"
          type="checkbox"
          value="26"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          京都府
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item26"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item26"
          type="checkbox"
          value="27"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          大阪府
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item27"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item27"
          type="checkbox"
          value="28"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          兵庫県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item28"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item28"
          type="checkbox"
          value="29"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          奈良県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item29"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item29"
          type="checkbox"
          value="30"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          和歌山県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item30"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item30"
          type="checkbox"
          value="31"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          鳥取県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item31"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item31"
          type="checkbox"
          value="32"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          島根県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item32"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item32"
          type="checkbox"
          value="33"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          岡山県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item33"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item33"
          type="checkbox"
          value="34"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          広島県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item34"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item34"
          type="checkbox"
          value="35"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          山口県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item35"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item35"
          type="checkbox"
          value="36"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          徳島県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item36"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item36"
          type="checkbox"
          value="37"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          香川県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item37"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item37"
          type="checkbox"
          value="38"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          愛媛県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item38"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item38"
          type="checkbox"
          value="39"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          高知県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item39"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item39"
          type="checkbox"
          value="40"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          福岡県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item40"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item40"
          type="checkbox"
          value="41"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          佐賀県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item41"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item41"
          type="checkbox"
          value="42"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          長崎県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item42"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item42"
          type="checkbox"
          value="43"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          熊本県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item43"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item43"
          type="checkbox"
          value="44"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          大分県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item44"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item44"
          type="checkbox"
          value="45"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          宮崎県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item45"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item45"
          type="checkbox"
          value="46"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          鹿児島県
        </span>
      </label>
    </li>
    <li>
      <label
        class="_checkbox_1py7j_1"
        for="item46"
      >
        <input
          class="_checkbox__input_1py7j_10"
          id="item46"
          type="checkbox"
          value="47"
        />
        <span
          class="_checkbox__box_1py7j_19 "
        >
          <span
            class="_checkbox__check_1py7j_30"
          />
        </span>
        <span
          class="_checkbox__label_1py7j_41 "
        >
          沖縄県
        </span>
      </label>
    </li>
  </ul>
</div>
`;
